compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/AnnotationDesugar.java

```diff
@@ -191,6 +199,84 @@ private void defineClassAnnotations(BLangPackage pkgNode, SymbolEnv env, BLangFu
         }
     }
 
+    private boolean isServiceDeclaration(BLangClassDefinition serviceClass) {
+        return Objects.nonNull(serviceClass.getFlags())
+                && serviceClass.getFlags().contains(Flag.SERVICE)
+                && (serviceClass.isServiceDecl || serviceClass.getFlags().contains(Flag.OBJECT_CTOR));
+    }
+
+    private void addIntrospectionInfoAnnotation(BLangClassDefinition serviceClass) {
+        Location position = serviceClass.pos;
+
+        // Create Annotation Attachment.
+        BLangAnnotationAttachment annoAttachment = (BLangAnnotationAttachment) TreeBuilder.createAnnotAttachmentNode();
+        final SymbolEnv pkgEnv = symTable.pkgEnvMap.get(serviceClass.symbol.getEnclosingSymbol());
+        BSymbol annSymbol = symResolver.lookupSymbolInAnnotationSpace(
+                pkgEnv, names.fromString(SERVICE_INTROSPECTION_INFO_ANN));
+        if (annSymbol instanceof BAnnotationSymbol) {
+            annoAttachment.annotationSymbol = (BAnnotationSymbol) annSymbol;
+        }
+
+        BLangIdentifier identifierNode = (BLangIdentifier) TreeBuilder.createIdentifierNode();
+        annoAttachment.annotationName = identifierNode;
+        annoAttachment.annotationName.value = SERVICE_INTROSPECTION_INFO_ANN;
+        annoAttachment.pos = position;
+        annoAttachment.annotationName.pos = position;
+
+        // create record literal related to annotation node
+        BLangRecordLiteral literalNode = (BLangRecordLiteral) TreeBuilder.createRecordLiteralNode();
+        annoAttachment.expr = literalNode;
+        BLangIdentifier pkgAlias = (BLangIdentifier) TreeBuilder.createIdentifierNode();
+        pkgAlias.setValue(BUILTIN_PKG_KEY);
```

Comment:
We don't have a builtin package now. We need to use the correct module.

Reply:
For compiler generated annotations (`DefaultableArgs` annotation etc.) `ballerina/builtin` package has been used. Since, this is also a compiler generated one we have added it.

Reply:
I think that just hasn't been updated with the builtin change.

Reply:
Since we are attaching this annotation for any service-declarations what would be the correct package name to be used ?

Reply:
This pkgAlias seems to be for the module in which the annotation is defined. So in this case I guess it would have to be the lang.annotations module.